### PR TITLE
Demo issuer partial generated cli

### PR DESCRIPTION
**Pull request by filippo-tenaglia**
This PR will contain the source code of the demo created using the openapi generator tool with the flag typescript.
The folder contains many files autogenerated as described on the readme.md found on the parent folder and few files that should integrated the cli's logic.
These files are:
main.Cli.ts and signer-cli.ts.
At this moment if you tried the code the terminal return this error after the calling to the first API implemented:
ReferenceError: XMLHttpRequest is not defined

Can @gunzip  help me to understand where is the error?

Here its are the steps to reproduce it:
1) npm install
2) tsc
3) npm run start-cli
4) select "signer"
5) insert any character tho execute the call.

Tho exit from the cli you can choose any other APIs.

### CODE REVIEW

**Comment by gunzip** [comment]
@filippo-tenaglia the generated code is utilizing the [whatwg-fetch fetch](https://www.npmjs.com/package/whatwg-fetch) package; it's designed for web browsers, which is the cause of the problem. To make the generator build for NodeJS instead, you probably need to feed it some sort of command line flag. Let me review the other PR with the ADR to make sure we're aligned on the code generating tool before moving on.

**Comment by filippo-tenaglia** [comment]
@gunzip  I resolved the problem updating node to the latest LTS version (18.14) now the API response with a 422 error code but response!

**Comment by filippo-tenaglia** [comment]
@gunzip  I solved the issue of the empty body. Now I'm trying to refactor the code because on SignerCli.ts there are too many promises!

**Comment by filippo-tenaglia** [comment]
@gunzip  in this last commit I added all references already tested on ts-cli probject so now you can call all APIs correctly. There is, probably, an error during the creation of a new signature request because the class SignatureField autogenerated seems have to need both the coords and the unique name! so, If you want to try to create a new signature request, select the unique name option during the creation of the clauses.

**Comment by filippo-tenaglia** [comment]
@lucacavallaro  and @grausof  I modified the .gitignore file to add on the repository the folder models autogenerated by openapi generator

File: docs/demo-issuer/openapi-gen-ts/src/generated/models/SignatureField.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/**
```
Comment (filippo-tenaglia) [inline docs/demo-issuer/openapi-gen-ts/src/generated/models/SignatureField.ts:1]: @gunzip  @lucacavallaro  @grausof  using this class seems that I must pass both unique_name and attrs with some values, can you see if it is correct or I didn't see something?
The consequence of this behaviour can create problems on the backend right?

**Comment by gunzip** [comment]
closed in favor of https://github.com/pagopa/io-sign/pull/91

**Pull request closed without merging**